diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ src/ros2/rviz/rviz_ogre_vendor/CMakeLists.txt
@@ -203,6 +203,8 @@ macro(build_ogre)
       -DOGRE_BUILD_COMPONENT_CSHARP:BOOL=FALSE
       -DOGRE_BUILD_COMPONENT_MESHLODGENERATOR:BOOL=FALSE
       -DOGRE_BUILD_COMPONENT_BITES:BOOL=FALSE
+      -DOGRE_BUILD_PLUGIN_DOT_SCENE:BOOL=FALSE
+      -DOGRE_BUILD_TOOLS:BOOL=FALSE
       ${extra_cmake_args}
       -Wno-dev
     PATCH_COMMAND
